types/firestore.d.ts

```diff
@@ -724,6 +765,10 @@ declare namespace FirebaseFirestore {
      * This function returns a new (immutable) instance of the Query (rather
      * than modify the existing instance) to impose the field mask.
      *
+     * @deprecated Use `get(options:GetOptions)` or `stream(options:GetOptions)`
+     * to specify a field mask. The field mask specified via `select()` does not
+     * apply to `getAll()`/`streamAll()`.
```

Comment:
Er, yikes.  Deprecating select() in order to support this new use case seems like the tail wagging the dog a bit.